### ISSUE TITLE
White Screen after every Boot, maybe because of newest Kernel? 

### ISSUE THREAD

**Issue by phenomeus** [raspiblitz/raspiblitz#1436]
Hi everybody, 

I have a white screen that is not going away, i think its dead (but still powered on?) or maybe that came after kernel updates etc. can‘t say for sure, i noticed it too late. 

`uname -a:
Linux raspberrypi 5.4.51-v7l+ #1327 SMP Thu Jul 23 11:04:39 BST 2020 armv7l GNU/Linux `

Any idea how to debug and get it to work again? I already tried to power down the pi and removed the power cord. 

Thanks in advance

**Comment by phenomeus** [comment]
It looks like a kernel and driver issue 

https://github.com/goodtft/LCD-show/issues/223

**Comment by phenomeus** [comment]
There is a fork that is updated to work with newest kernel, trying right now myself ~~, will report back.~~ its working. Reapplying the screen rotation again  

```
sudo rm -rf LCD-show
git clone https://github.com/MrYacha/LCD-show.git
chmod -R 755 LCD-show
cd LCD-show/
sudo ./LCD35-show
```

**Comment by phenomeus** [comment]
Rotation activation by menu brings back white screen. Will debug further

**Comment by phenomeus** [comment]
I created two pull requests. My screen is working again and is rotated correctly. 

(Sorry for 2 patches, I’m still learning how to use github ;) )

**Comment by phenomeus** [comment]
I have to rotate the touchscreen also, 180 degree

**Comment by phenomeus** [comment]
Touch screen rotation also fixed and tested. Logic had to be turned around. PR is online

**Comment by rootzoll** [comment]
Thanks .. hope to find time to review your changes in the next days.

**Comment by frennkie** [comment]
I have this issue too.

**Comment by frennkie** [comment]
@phenomeus wouldn't the three PRs better be put into one..? Would be great if you could look into that.

Would also be nice to have a more descriptive branch name... Then it would be easier for me to quickly check out yand test your proposed changes.

**Comment by phenomeus** [comment]
Totally agree but I’m a GitHub newb :) seriously no idea how to merge my 3 prs and work with branches. 

Open to any guidance.

**Comment by frennkie** [comment]
But you do use the command line, right?!

In general I would recommend you go have a look some tutorials.. but here are the essential steps:

```
git remote add phenomeus https://github.com/phenomeus/raspiblitz.git
git remote add upstream https://github.com/rootzoll/raspiblitz.git
git fetch --all --prune
checkout -b fix-white-screen upstream/v1.6
git merge phenomeus/patch-2
git merge phenomeus/patch-3
git merge phenomeus/patch-4
git push -u phenomeus fix-white-screen
```

**Comment by frennkie** [comment]
Are actually all changes needed? Or is this sufficient:

Step 1) Update kernel 

```
sudo apt update
sudo apt full-upgrade
/home/admin/XXshutdown.sh reboot  # not sure if needed
```

Step 2) 

```
sudo sed -i "s/^dtoverlay=.*/dtoverlay=waveshare35a:rotate=90/g" /boot/config.txt
rm -rf /home/admin/LCD-show
git clone https://github.com/MrYacha/LCD-show.git /home/admin/LCD-show
cd /home/admin/LCD-show/
sudo ./LCD35-show  # this will reboot
```

**Comment by phenomeus** [comment]
Thanks for the input. 

First you have to use a fork for lcdshow 

Than modify to waveshare35a

Than fix rotation logic

**Comment by frennkie** [comment]
> Thanks for the input.

Will you consolidate the PRs? Or should we do it?

**Comment by ChuckNorrison** [comment]
testing this right now

// Can confirm, this issues hits me after rpi-update and as already found the kernel was updated to 5.4.51. 

// need ` chmod +x LCD35-show` to execute that script after git clone
-> execute prints: -> `mkdir: cannot create directory ‘/etc/X11/xorg.conf.d’: File exists`
Is this bad? Iam using the recommended screen which works fine till the rpi-update.
```
admin@raspberrypi /etc/X11/xorg.conf.d $ cat 99-calibration.conf 
Section "InputClass"
        Identifier      "calibration"
        MatchProduct    "ADS7846 Touchscreen"
        Option  "Calibration"   "3936 227 268 3880"
        Option  "SwapAxes"      "1"
EndSection
```

// after reboot from LCD35-show, screen is still not working

// selected the LCD-Rotate setting from menu after update lcdrotate.sh, still no screen after reboot.

// i started the SCREEN from menu. Flashing up some errors, i took a screen here:
```
# calibrating touchscreen ...
error='not installed'
Unable to connect to X server
error='aborted'
Starting the main menu ...
```
// /config-scripts/blitz.lcd.sh seems to be updated to waveshare35a too

// updated all the files manually cause iam afraidto set this as branch. still need to get into all this shell scripts

// does not work for me, i dont know

**Comment by phenomeus** [comment]
> > Thanks for the input.
> 
> Will you consolidate the PRs? Or should we do it?

Please do it, pretty busy right now

**Comment by rootzoll** [comment]
OK i implemented it now that way that on boot it will check if the kernel was updated and may correct the LCD driver then.

Also the lcd scripts now should check what kernal is running and use the correct rotation setting.

I was not able to test the kernel upgrade because `sudo apt-get upgrade` resulted in:
```
E: Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/w/webkit2gtk/libjavascriptcoregtk-4.0-18_2.28.3-2~deb10u1+rpi1_armhf.deb  404  Not Found [IP: 93.93.128.193 80]
```

So final testing is still TODO ... with upgrading kernel to see if it automatically adapts and to see if screen rotation is then working correctly.

**Comment by frennkie** [comment]
404 errors on `apt-get upgrade`...? Either the source has a - yet unnoticed - (temporary or permanent) problem or you forgot to run `apt-get update`.

**Comment by rootzoll** [comment]
@frennkie ah ok I tried `apt-get update` again and now the upgrade is working. Not sure why it was not working the first time. thanks.

**Comment by rootzoll** [comment]
OK this looks good now. When I upgrade the kernel on a fresh v1.6.0 and then `restart` it picks up on boot that the kernel has changed and updates the drivers.

The touchchscreen rotation gets messed up on LCD rotation ... maybe make a v1.7 issue to automatically call "calibration" after turning on LCD or doing rotation.

Last thing to test before release: If then a v1.6.1 (that will still have the old kernel because base sd card image will be changed on service release) will run thru good on recovering with touch screen.

**Comment by rootzoll** [comment]
Recoverings with v1.6.1 and Touchscreen went well. Closing issue for v1.6.1 release.

**Comment by uninflatable** [comment]
Hi, I updated today my rasiplitz from 1.6.0 to 1.6.1 using the image: https://raspiblitz.org/raspiblitz-v1.6.1-2020-10-20.img.gz

after installing 1.6.1 I get a white screen and with SSH I can't connect any longer. Please advise

**Comment by ChuckNorrison** [comment]
@uninflatable you need to recreate your sd card and boot from it regulary to get into the recovery process.

**Comment by uninflatable** [comment]
@ChuckNorrison excellent - this worked fine! Thank you for your support!

**Comment by ChuckNorrison** [comment]
today i saw the white screen again after some weeks untouched. my blitz worked yesterday. I was running 1.6.2 on this. Now i will reflash on 1.6.3

// after flashing 1.6.3 iam back at Control-D after phase 1 with default admin credentials was done and first reboot. My Blitz seems very unstable. Maybe hardware related... iam outta here, thanks anyway... i tried hard on this

// no control D spotted anymore after micro SD replacement and current release 1.7.0 RC2

**Comment by phenomeus** [comment]
Little headsup, I'm using this fork https://github.com/sun-23/LCD-show for kernel 5.10 

just in case you want to update here :)

**Comment by Mogultalk** [comment]
> @uninflatable you need to recreate your sd card and boot from it regulary to get into the recovery process.

How??
